core/src/main/java/com/radixdlt/consensus/bft/processor/BFTQuorumAssembler.java

```diff
@@ -177,39 +156,78 @@ private void processVoteInternal(Vote vote) {
     switch (this.pendingVotes.insertVote(vote)) {
       case VoteAccepted ignored -> log.trace("Vote has been processed but didn't form a quorum");
       case VoteRejected voteRejected -> log.trace(
-          "Vote has been rejected because of: {}", voteRejected.getReason());
-      case QuorumReached quorumReached -> {
-        this.hasReachedQuorum = true;
-        roundQuorumReachedEventDispatcher.dispatch(
-            new RoundQuorumReached(quorumReached.getRoundVotingResult(), vote));
-      }
+          "Vote has been rejected because of: {}", voteRejected.reason());
+      case QuorumReached quorumReached -> this.processQuorum(quorumReached.roundQuorum(), vote);
     }
 
     metrics.bft().successfullyProcessedVotes().inc();
   }
 
-  @Override
-  public void processProposal(Proposal proposal) {
-    forwardTo.processProposal(proposal);
-  }
+  private void processQuorum(RoundQuorum roundQuorum, Vote lastVote) {
+    if (hasRegularQuorumBeenFormedForCurrentRound()) {
```

Comment:
For timeout certificates, we do: "but we still replace our `bestQuorumFormedSoFarInCurrentRound` because(arguably) more signatures == a more trustworthy certificate."

But not for QCs - because we handle them immediately. That's fine, but maybe a little strange we care about doing it for TCs 🤷.

So I'd also support removing the logic to "improve" the timeout certificate, up to you!